### ISSUE TITLE
[Bug]: Can't access folder which name contain "[1-0]" on Web

### ISSUE THREAD

**Issue by Myuki** [nextcloud/server#32518]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I first try name `[2018-3-23]First-Post` and find the problem.

### Steps to reproduce

1. Create a folder name `[1-0]` or change a folder name to that.
2. Right click the folder on Web.
3. Nothing change on web and keep in the current folder.

### Expected behavior

Enter the folder.

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - bruteforcesettings: 2.4.0
  - cloud_federation_api: 1.7.0
  - dav: 1.22.0
  - federatedfilesharing: 1.14.0
  - files: 1.19.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - oauth2: 1.12.0
  - provisioning_api: 1.14.0
  - settings: 1.6.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - viewer: 1.8.0
  - workflowengine: 2.6.0
Disabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - admin_audit
  - circles: 24.0.0
  - comments: 1.14.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - encryption
  - federation: 1.14.0
  - files_external
  - files_pdfviewer: 2.5.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - notifications: 2.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - privacy: 1.8.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - text: 3.5.1
  - theming: 1.15.0
  - user_ldap
  - user_status: 1.4.0
  - weather_status: 1.4.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
No new log when I reproduce the bug.
```

### Additional info

Firefox Console:
```
Uncaught SyntaxError: invalid range in character class
    setDirectory https://example.com/js/files/merged-index.js?v=2ffb557d:495
    _setCurrentDir https://example.com/js/files/merged-index.js?v=2ffb557d:6188
    changeDirectory https://example.com/js/files/merged-index.js?v=2ffb557d:6116
    actionHandler https://example.com/js/files/merged-index.js?v=2ffb557d:3625
    action https://example.com/js/files/merged-index.js?v=2ffb557d:3040
    _onClickFile https://example.com/js/files/merged-index.js?v=2ffb557d:4974
    ge https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    dispatch https://example.com/dist/core-common.js?v=2ffb557d:2
    handle https://example.com/dist/core-common.js?v=2ffb557d:2
    add https://example.com/dist/core-common.js?v=2ffb557d:2
    j https://example.com/dist/core-common.js?v=2ffb557d:2
    n https://example.com/dist/core-common.js?v=2ffb557d:2
    kt https://example.com/dist/core-common.js?v=2ffb557d:2
    each https://example.com/dist/core-common.js?v=2ffb557d:2
    each https://example.com/dist/core-common.js?v=2ffb557d:2
    kt https://example.com/dist/core-common.js?v=2ffb557d:2
    on https://example.com/dist/core-common.js?v=2ffb557d:2
    initialize https://example.com/js/files/merged-index.js?v=2ffb557d:4437
    FileList https://example.com/js/files/merged-index.js?v=2ffb557d:4080
    initialize https://example.com/js/files/merged-index.js?v=2ffb557d:87
    <anonymous> https://example.com/js/files/merged-index.js?v=2ffb557d:405
    Se https://example.com/dist/core-common.js?v=2ffb557d:2
    setTimeout handler*64492/Se< https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    ge https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    <anonymous> https://example.com/js/files/merged-index.js?v=2ffb557d:404
    EventListener.handleEvent* https://example.com/js/files/merged-index.js?v=2ffb557d:401
merged-index.js:495:33
    search self-hosted:3916
    setDirectory https://example.com/js/files/merged-index.js?v=2ffb557d:495
    _setCurrentDir https://example.com/js/files/merged-index.js?v=2ffb557d:6188
    changeDirectory https://example.com/js/files/merged-index.js?v=2ffb557d:6116
    actionHandler https://example.com/js/files/merged-index.js?v=2ffb557d:3625
    action https://example.com/js/files/merged-index.js?v=2ffb557d:3040
    _onClickFile https://example.com/js/files/merged-index.js?v=2ffb557d:4974
    ge https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    dispatch https://example.com/dist/core-common.js?v=2ffb557d:2
    handle https://example.com/dist/core-common.js?v=2ffb557d:2
    (Async: EventListener.handleEvent)
    add https://example.com/dist/core-common.js?v=2ffb557d:2
    j https://example.com/dist/core-common.js?v=2ffb557d:2
    n https://example.com/dist/core-common.js?v=2ffb557d:2
    kt https://example.com/dist/core-common.js?v=2ffb557d:2
    each https://example.com/dist/core-common.js?v=2ffb557d:2
    each https://example.com/dist/core-common.js?v=2ffb557d:2
    kt https://example.com/dist/core-common.js?v=2ffb557d:2
    on https://example.com/dist/core-common.js?v=2ffb557d:2
    initialize https://example.com/js/files/merged-index.js?v=2ffb557d:4437
    FileList https://example.com/js/files/merged-index.js?v=2ffb557d:4080
    initialize https://example.com/js/files/merged-index.js?v=2ffb557d:87
    <anonymous> https://example.com/js/files/merged-index.js?v=2ffb557d:405
    Se https://example.com/dist/core-common.js?v=2ffb557d:2
    (Async: setTimeout handler)
    Se https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    ge https://example.com/dist/core-common.js?v=2ffb557d:2
    i https://example.com/dist/core-common.js?v=2ffb557d:2
    <anonymous> https://example.com/js/files/merged-index.js?v=2ffb557d:404
    (Async: EventListener.handleEvent)
    <anonymous> https://example.com/js/files/merged-index.js?v=2ffb557d:401
```

**Comment by solracsf** [comment]
Should be fixed in v24.0.1. Please upgrade, test and report back.

**Comment by Myuki** [comment]
It is fixed in v24.0.1.
